src/helper-modules/companies-house/known-errors.ts

```diff
@@ -0,0 +1,33 @@
+import { AxiosError } from 'axios';
+
+import { CompaniesHouseInvalidAuthorizationException } from './exception/companies-house-invalid-authorization.exception';
+import { CompaniesHouseMalformedAuthorizationHeaderException } from './exception/companies-house-malformed-authorization-header.exception';
+import { CompaniesHouseNotFoundException } from './exception/companies-house-not-found.exception';
+
+export type KnownErrors = KnownError[];
+
+type KnownError = { caseInsensitiveSubstringToFind: string; throwError: (error: AxiosError) => never };
+
+export const getCompanyMalformedAuthorizationHeaderKnownCompaniesHouseError = (): KnownError => ({
+  caseInsensitiveSubstringToFind: 'Invalid Authorization header',
+  throwError: (error) => {
+    throw new CompaniesHouseMalformedAuthorizationHeaderException(
+      `Invalid 'Authorization' header. Check that your 'Authorization' header is well-formed.`,
+      error,
+    );
+  },
+});
+
+export const getCompanyInvalidAuthorizationKnownCompaniesHouseError = (): KnownError => ({
+  caseInsensitiveSubstringToFind: 'Invalid Authorization',
+  throwError: (error) => {
+    throw new CompaniesHouseInvalidAuthorizationException('Invalid authorization. Check your Companies House API key.', error);
+  },
+});
+
+export const getCompanyNotFoundKnownCompaniesHouseError = (registrationNumber: string): KnownError => ({
```

Comment:
As MDM and DTFS use more specific error string rather than status code error matching, I would need to reproduce 429 and 500 errors from CH API to ascertain the error strings associated with these errors (which may be difficult), as these errors are not documented in the API documentation: https://developer-specs.company-information.service.gov.uk/companies-house-public-data-api/reference/company-profile/company-profile

As we don't specifically need to handle these errors (we just return a 500 for our endpoint for both), I don't think we need specific error handling, as in MDM we log Axios requests/responses anyway.